resolwe_bio/processes/import_data/sra_file.py

```diff
@@ -0,0 +1,403 @@
+    sequencing platforms.
+
+    """
+    slug = 'import-sra-single'
+    name = 'SRA data (single-end)'
+    process_type = 'data:reads:fastq:single'
+    version = '1.0.0'
+    category = 'Import'
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.RAW
+    entity = {
+        'type': 'sample',
+        'descriptor_schema': 'sample',
+    }
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image': 'resolwebio/common:1.3.1'
+            }
+        },
+        'resources': {
+            'cores': 1,
+            'memory': 1024,
+            'network': True,
+        },
+    }
+    data_name = '{{ sra_accession|first }}'
+
+    class Input:
+        """Input fields to process ImportSraSingle."""
+        sra_accession = ListField(StringField(), label='SRA accession(s)')
+        show_advanced = BooleanField(label='Show advanced options', default=False)
+
+        class Advanced:
+            """Advanced options."""
+            min_spot_id = IntegerField(label='Minimum spot ID', required=False)
+            max_spot_id = IntegerField(label='Maximum spot ID', required=False)
+            min_read_len = IntegerField(label='Minimum read length', required=False)
+            clip = BooleanField(label='Clip adapter sequences', default=False)
+            aligned = BooleanField(label='Dump only aligned sequences', default=False)
+            unaligned = BooleanField(label='Dump only unaligned sequences', default=False)
+
+        advanced = GroupField(Advanced, label='Advanced options', hidden='!show_advanced')
+
+    class Output:
+        """Output fields to process ImportSraSingle."""
+        fastq = ListField(FileField(), label='Reads file')
+        fastqc_url = ListField(
+            FileHtmlField(),
+            label='Quality control with FastQC',
+        )
+        fastqc_archive = ListField(FileField(), label='Download FastQC archive')
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        for srr in inputs.sra_accession:
+            cmd = Cmd['fastq-dump']['--gzip']
+            if inputs.advanced.min_spot_id:
+                cmd = cmd['--minSpotId'][inputs.advanced.min_spot_id]
+            if inputs.advanced.max_spot_id:
+                cmd = cmd['--maxSpotId'][inputs.advanced.max_spot_id]
+            if inputs.advanced.min_read_len:
+                cmd = cmd['--minReadLen'][inputs.advanced.min_read_len]
+            if inputs.advanced.clip:
+                cmd = cmd['--clip']
+            if inputs.advanced.aligned:
+                cmd = cmd['--aligned']
+            if inputs.advanced.unaligned:
+                cmd = cmd['--unaligned']
+            return_code, _, _ = cmd[srr] & TEE(retcode=None)
+            if return_code:
+                self.error(f'Download of {srr} reads with fastq-dump failed.')
+
+        fastq_gz = glob.glob('*.fastq.gz')
```

Comment:
You can use `pathlib.glob` function to gather files with a specific pattern. There are a number of occurrences throughout the document.